### PR TITLE
Added Decentlab devices

### PR DESCRIPTION
**Pull request by decentlab**
Added the following Decentlab device decoders:
- DL-ALB
- DL-ATM22
- DL-BLG
- DL-CTD10
- DL-CWS
- DL-DLR2-002
- DL-DLR2-003
- DL-DLR2-004
- DL-DLR2-005
- DL-DLR2-006
- DL-DLR2-008
- DL-DLR2-009
- DL-DLR2-010
- DL-DLR2-011
- DL-DLR2-012
- DL-DS18
- DL-GMM
- DL-ISF
- DL-ITST
- DL-KL66
- DL-LID
- DL-LPW
- DL-LWS
- DL-NTU
- DL-OPTOD
- DL-PAR
- DL-PHEHT
- DL-PM
- DL-PR36CTD
- DL-PYR
- DL-RHC
- DL-SDD
- DL-TP
- DL-TRS11
- DL-TRS12
- DL-TRS21
- DL-WRM
- DL-ZN1
- DL-ZN2

### CODE REVIEW

File: types/decentlab/DLR2-008/uplink.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+function consume(event) {
+  const payload = event.data.payloadHex;
+
+  if (event.device !== undefined) {
+    if (event.device.customFields !== undefined) {
+      const { customFields } = event.device;
+
+      if (customFields.R !== undefined) {
```
Comment (J2thatsme) [inline types/decentlab/DLR2-008/uplink.js:98]: Please write out the custom field parameter R

File: types/decentlab/DLR2-009/uplink.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+function consume(event) {
+  const payload = event.data.payloadHex;
+
+  if (event.device !== undefined) {
+    if (event.device.customFields !== undefined) {
+      const { customFields } = event.device;
+
+      if (customFields.R !== undefined) {
```
Comment (J2thatsme) [inline types/decentlab/DLR2-009/uplink.js:102]: Please write out the custom field parameter R

File: types/decentlab/KL66/uplink.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+  const payload = event.data.payloadHex;
+
+  if (event.device !== undefined) {
+    if (event.device.customFields !== undefined) {
+      const { customFields } = event.device;
+
+      if (customFields.f0 !== undefined) {
+        decentlab_decoder.PARAMETER.f0 = customFields.f0;
+      }
+      if (customFields.k !== undefined) {
```
Comment (J2thatsme) [inline types/decentlab/KL66/uplink.js:120]: Please write out the custom field parameter f0 and k

File: types/decentlab/PR36CTD/uplink.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+  const payload = event.data.payloadHex;
+
+  if (event.device !== undefined) {
+    if (event.device.customFields !== undefined) {
+      const { customFields } = event.device;
+
+      if (customFields.kp !== undefined) {
+        decentlab_decoder.PARAMETER.kp = customFields.kp;
+      }
+      if (customFields.kec !== undefined) {
```
Comment (J2thatsme) [inline types/decentlab/PR36CTD/uplink.js:114]: Please write out the custom field parameter kp and kec

**Review by J2thatsme — COMMENTED** [review]
Thanks for your pull request.
I commented some small adjustments but it looks good overall.

File: types/decentlab/DLR2-004/uplink.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+function consume(event) {
+  const payload = event.data.payloadHex;
+
+  if (event.device !== undefined) {
+    if (event.device.customFields !== undefined) {
+      const { customFields } = event.device;
+
+      if (customFields.R !== undefined) {
```
[the remark this answers is not in the document]
Comment (decentlab) [reply types/decentlab/DLR2-004/uplink.js:98]: > Please write out the custom field parameter R

Thanks for the review. By write out, do you mean to include the error topic?

[the remark this answers is not in the document]
Comment (J2thatsme) [reply types/decentlab/DLR2-004/uplink.js:98]: R is not really human readable. Customfields are shown in our frontend, so if its standing for Radius for example just call it that.

[the remark this answers is not in the document]
Comment (decentlab) [reply types/decentlab/DLR2-004/uplink.js:98]: I see. `R` in this case, is meant `shuntResistor`. However, the users receive the device with the information of parameter `R`. They wouldn't know anything of `shuntResistor`.

So somehow we have to include `R`. Does it make sense to use `R` or `R_shuntResistor`?

[the remark this answers is not in the document]
Comment (J2thatsme) [reply types/decentlab/DLR2-004/uplink.js:98]: R_shuntResistor sounds good 👍

**Review by J2thatsme — APPROVED** [review]
Thanks for the adjustments.
I will merge it now and it should be available shortly on Akenza.

**Pull request merged**